packages/charts/src/charts/line/utils/data.utils.ts

```diff
@@ -0,0 +1,130 @@
+import { stack } from 'd3-shape';
+
+import { KeyNamesValuesType } from '../types';
+
+export const calculateStackData = (
+  data: Record<string, any>[],
+  labelSelector: string,
+  keys: string[]
+): any[] => {
+  const stackedData = stack().keys(keys)(data);
+
+  const newData = stackedData.reduce((acc, value) => {
+    const stackValues = value.map((el, idx) => {
+      const base = {
+        ...acc[idx],
+        [labelSelector]: el.data[labelSelector],
+      };
+      const first = {
+        ...base,
+        [value.key]: el[1],
+      };
+      return first;
+    });
+
+    return [...stackValues];
+  }, []);
+
+  if (!keys.length) {
+    return data.map((el) => ({ labelSelector: el[labelSelector] }));
+  }
+
+  return newData;
+};
+
+export const calculateStackAreaData = (
+  data: Record<string, any>[],
+  labelSelector: string,
+  keys: string[]
+): any => {
+  const stackedData = stack().keys(keys)(data);
+
+  const reduceStack = (id: number) => (
+    acc: Record<string, any>[],
+    value: Record<string, any>
+  ) => {
+    const stackValues = value.map((el: Record<string, any>, idx: number) => {
+      const base = {
+        ...acc[idx],
+        [labelSelector]: el.data[labelSelector],
+      };
+      const first = {
+        ...base,
+        [value.key]: el[id],
+      };
+      return first;
+    });
+
+    return [...stackValues];
+  };
+
+  const firstDataPart = stackedData.reduce(reduceStack(0), []);
+  const secondDataPart = stackedData.reduce(reduceStack(1), []);
+
+  if (!keys.length) {
+    return data.map((el) => ({ labelSelector: el[labelSelector] }));
+  }
+
+  return { firstDataPart, secondDataPart: secondDataPart.reverse() };
+};
+
+export const sortKeys = (data: Record<string, any>[], keys: string[]) => {
```

Comment:
- Move it to @keen.io/charts-utils
- Consider renaming the function + add comments